libsolidity/Types.cpp

```diff
@@ -452,6 +452,17 @@ unsigned FunctionType::getSizeOnStack() const
 	}
 }
 
+std::string FunctionType::getCanonicalSignature() const
+{
+	auto parameters = getParameterTypes();
+	std::string ret = "NAME("; //TODO: how to get function name from FunctionType
```

Comment:
As discussed: The function type should not know about the name of its function, so it has to be prepended from a different place.